script/utils/update-redirects-data.ts

```diff
@@ -21,14 +31,18 @@ const updateRedirectsData = async(currentBook: BookWithOSWebData, newBook: BookW
   const redirects: RedirectsData = fs.existsSync(redirectsBookPath) ? await import(redirectsBookPath) : [];
 
   const flatCurrentTree = flattenArchiveTree(currentBook.tree).filter((section) => section.id !== currentBook.id);
-  const flatNewTree = flattenArchiveTree(newBook.tree).filter((section) => section.id !== currentBook.id);
+  const currentSections = flatCurrentTree.filter(archiveTreeSectionIsPage);
+  const flatNewTree = allowBookRedirect
+    ? flattenArchiveTree(newBook.tree).filter((section) => section.id !== newBook.id)
+    : flattenArchiveTree(newBook.tree).filter((section) => section.id !== currentBook.id);
 
-  const formatSection = (section: LinkedArchiveTreeNode) => ({
-    bookId: currentBook.id,
-    pageId: section.id,
+  const formatSection = (section: LinkedArchiveTreeNode, newSection?: ArchiveTreeNode) => ({
+    bookId: allowBookRedirect ? newBook.id : currentBook.id,
+    pageId: newSection?.id ? stripIdVersion(newSection.id) : section.id,
     pathname: decodeURI(
       content.getUrl({ book: { slug: currentBook.slug }, page: { slug: section.slug } })
     ),
+    ...(allowBookRedirect && {query: `?${messageQueryParameterName}=retired`}),
```

Comment:
@TomWoodward where would be the best place to should store possible `message` values?